lib/systemd_testsuite_test.pm

```diff
@@ -123,7 +123,13 @@ sub post_fail_hook {
     my ($self) = @_;
     #upload logs from given testname
     $self->tar_and_upload_log('/var/opt/systemd-tests/logs', '/tmp/systemd_testsuite-logs.tar.bz2');
-    $self->save_and_upload_log('journalctl --no-pager -axb -o short-precise', 'journal.log');
+    if ((script_run 'test -d /var/log/journal')) {
+        $self->tar_and_upload_log('/var/log/journal', 'binary-journal-log.tar.bz2');
```

Comment:
Why not both in the same tar call? 
```suggestion
        $self->tar_and_upload_log('/var/log/journal /run/log/journal', 'binary-journal-log.tar.bz2');
```

Reply:
Indeed, that looks more elegant and it seems the code doesn't error out when one of the directories isn't present.
I've updated the pr, thanks.